jwst/ami/ami_analyze_step.py

```diff
@@ -38,31 +38,36 @@ def process(self, input):
         self.log.info('Initial rotation guess = %g deg', rotate)
 
         # Open the input data model
-        with datamodels.ImageModel(input) as input_model:
-
-            # Get the name of the filter throughput reference file to use
-            self.filter_ref_name = self.get_reference_file(input_model,
-                                   'throughput')
-            self.log.info('Using filter throughput reference file %s',
-                           self.filter_ref_name)
-
-            # Check for a valid reference file
-            if self.filter_ref_name == 'N/A':
-                self.log.warning('No THROUGHPUT reference file found')
-                self.log.warning('AMI analyze step will be skipped')
-                result = input_model.copy()
-                result.meta.cal_step.ami_analyze = 'SKIPPED'
-                return result
-
-            # Open the filter throughput reference file
-            filter_model = datamodels.ThroughputModel(self.filter_ref_name)
-
-            # Do the LG analysis on the input image
-            result = ami_analyze.apply_LG(input_model, filter_model,
-                                          oversample, rotate)
-
-            # Close the reference file and update the step status
-            filter_model.close()
-            result.meta.cal_step.ami_analyze = 'COMPLETE'
-
-        return result
+        try:
+            with datamodels.ImageModel(input) as input_model:
+
+                # Get the name of the filter throughput reference file to use
+                self.filter_ref_name = self.get_reference_file(input_model,
+                                       'throughput')
+                self.log.info('Using filter throughput reference file %s',
+                               self.filter_ref_name)
+
+                # Check for a valid reference file
+                if self.filter_ref_name == 'N/A':
+                    self.log.warning('No THROUGHPUT reference file found')
+                    self.log.warning('AMI analyze step will be skipped')
+                    result = input_model.copy()
+                    result.meta.cal_step.ami_analyze = 'SKIPPED'
+                    return result
```

Comment:
Here's an instance of the step skipping and returning a copy of the input. Does this get caught and handled appropriately up in the ``calwebb_ami3`` pipeline module? The rest of the pipeline is impossible to execute without output from this step.

Reply:
Hmm, good question.  It doesn't, as the pipeline has no clue that the step has been skipped.

Reply:
I will fix that in this PR.